pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -511,14 +511,30 @@ func SeedSettingOwnerChecksEnabled(settings *gardencorev1beta1.SeedSettings) boo
 	return settings == nil || settings.OwnerChecks == nil || settings.OwnerChecks.Enabled
 }
 
-// SeedSettingDependencyWatchdogEndpointEnabled returns true if the depedency-watchdog-endpoint is enabled.
-func SeedSettingDependencyWatchdogEndpointEnabled(settings *gardencorev1beta1.SeedSettings) bool {
-	return settings == nil || settings.DependencyWatchdog == nil || settings.DependencyWatchdog.Endpoint == nil || settings.DependencyWatchdog.Endpoint.Enabled
+// SeedSettingDependencyWatchdogWeederEnabled returns true if the dependency-watchdog-weeder is enabled. By default, this will be enabled unless explicitly set to false in `settings.DependencyWatchdog.Weeder.Enabled` or `settings.DependencyWatchdog.Endpoint.Enabled`(deprecated).
+// TODO:(himanshu-kun) Rely just on `settings.DependencyWatchdog.Weeder` field after few releases.
+func SeedSettingDependencyWatchdogWeederEnabled(settings *gardencorev1beta1.SeedSettings) bool {
+	if settings == nil || settings.DependencyWatchdog == nil {
+		return true
+	}
+	weederEnabled := settings.DependencyWatchdog.Weeder == nil || settings.DependencyWatchdog.Weeder.Enabled
+	// settings.DependencyWatchdog.Endpoint is deprecated and will be removed in the future release.
+	endpointEnabled := settings.DependencyWatchdog.Endpoint == nil || settings.DependencyWatchdog.Endpoint.Enabled
+	// (weederEnabled && endpointEnabled) is introduced with the assumption that both fields can't be set as that validation has been introduced.
+	return weederEnabled && endpointEnabled
```

Comment:
```suggestion
	weederEnabled := settings.DependencyWatchdog.Weeder != nil && settings.DependencyWatchdog.Weeder.Enabled
	// settings.DependencyWatchdog.Endpoint is deprecated and will be removed in the future release.
	endpointEnabled := settings.DependencyWatchdog.Endpoint != nil && settings.DependencyWatchdog.Endpoint.Enabled
	return weederEnabled || endpointEnabled
```
This looks more logical.

Reply:
thanks , I added the previous logic before I hadn't seen the defaulting code , thats why I tried handling the case of `settings.DependencyWatchdog.Weeder == nil` && `settings.DependencyWatchdog.Endpoint == nil`